webapp/src/pages/home/Home.js

```diff
@@ -1,207 +1,88 @@
 import React, { Component } from 'react';
-import logo from '../../images/indaba-logo-dark.png';
 import './Home.css';
-import { applicationFormService } from "../../services/applicationForm";
-import { offerServices } from "../../services/offer/offer.service";
 import { NavLink } from "react-router-dom";
-import { invitedGuestServices } from '../../services/invitedGuests/invitedGuests.service';
-import CookieConsent from "react-cookie-consent";
+import { eventService } from "../../services/events/events.service";
+import { organisationService } from "../../services/organisation/organisation.service";
 
-const DEFAULT_EVENT_ID = process.env.REACT_APP_DEFAULT_EVENT_ID || 1;
-
-const headings = ["Event", "Start date", "End date", "Status"];
-// const fieldNames = ["description", "start_date", "end_date", "status"]; unused
 
 class Home extends Component {
 
-  constructor(props) {
-    super(props);
-
-    this.state = {
-      headings: headings,
-      rows: [],
-      applicationStatus: null,
-      offer:null
+    constructor(props) {
+      super(props);
+  
+      this.state = {
+          events: null,
+          organisation: null,
+          errors: []
+      }
     }
-  }
-
-  componentDidMount() {
-    // Temporarily removed during campaign to get people to complete their applications
-    // getEvents().then(response => {
-    //   if (response) {
-    //     this.setState({
-    //       headings: headings,
-    //       rows: response
-    //     })
-    //   }
-    // });
-
-    // Only check for response/offer/invited guest if user is logged in
-    const user = localStorage.getItem('user');
-    if (!user)
-      return 
-
-    applicationFormService.getResponse(DEFAULT_EVENT_ID).then(resp => {
-      let applicationStatus = null;
-      if (resp.response) {
-        if (resp.response.is_submitted) {
-          applicationStatus = "Submitted";
-        }
-        else if (resp.response.is_withdrawn) {
-          applicationStatus = "Withdrawn";
+  
+    componentDidMount() {
+        if (this.props.user) {
+            eventService.getEvents().then(response=>{
+                if (response.error) {
+                    this.setState(prevState=>({
+                        errors: [...prevState.errors, response.error]
+                    }));
+                }
+                this.setState({
+                    events: response.events,
+                });
+            });
         }
-        else {
-          applicationStatus = "NOT Submitted";
+        
+        if (this.props.setEvent) {
```

Comment:
Why is this being done?

Reply:
The check is so the test doesn't break when setEvent is null. The calling setEvent with nulls is to "reset" the navbar in App.js when the user navigates away from a specific event to the organisation home page.